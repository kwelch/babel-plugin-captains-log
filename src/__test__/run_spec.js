import path from "path";
import fs from "fs";
import * as babel from "babel-core";
import PLUGIN from "../index";

const dirname = path.resolve(__dirname, "./__fixtures__");

export default function run_spec(name, options) {
  fs.readdirSync(dirname).forEach(filename => {
    testFile(dirname, filename, name, options);
  });
}

export function testFile(dirname, filename, testName, options, useBabelRC) {
  const source = read(path.resolve(dirname, filename));
  const output = transform(filename, source, options, useBabelRC);
  test(`${filename}-transform: ${testName}`, () => {
    expect(source + "~".repeat(80) + "\n" + output).toMatchSnapshot(filename);
  });
}

export function read(filename) {
  return fs.readFileSync(filename, "utf8");
}

export function transform(filename, source, options = {}) {
  const { code } = babel.transform(source, {
    filename: filename,
    babelrc: false,
    plugins: [[PLUGIN, options]],
  });
  return code;
}
