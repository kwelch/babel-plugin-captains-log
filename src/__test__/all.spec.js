import path from "path";
import { testFile } from "./run_spec.js";

testFile(path.resolve(__dirname, "__fixtures__"), "all.js", "all", null, true);
