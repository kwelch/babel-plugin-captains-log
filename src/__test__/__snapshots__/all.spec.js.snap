// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all.js 1`] = `
"console.log(a);
console.log(a.length);
console.log(a.map());

function add() {
  console.log(\\"1\\");
}

const subtract = () => {
  console.log(\\"2\\");
};

let multiply;
multiple = () => {
  console.log(\\"3\\");
};

let divide;
divide = function() {
  console.log(\\"4\\");
};

const power = function pow() {
  console.log(5);
};

const obj = {
  method1: function() {
    console.log(6);
  },
  method2: () => {
    console.log(7);
  },
  method3() {
    console.log(8);
  },
};

const Component = () => {
  const privateMethod = () => {
    console.log(1);
  };
  return class {
    showList = () => {
      arr.map(i => {
        console.log(i);
      });
    };
    render() {
      console.log(2);
    }
  };
};
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
\\"use strict\\";

var _classCallCheck2 = require(\\"babel-runtime/helpers/classCallCheck\\");

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require(\\"babel-runtime/helpers/createClass\\");

var _createClass3 = _interopRequireDefault(_createClass2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

console.log(\\"all.js(1:0)\\", \\"a\\", a);
console.log(\\"all.js(2:0)\\", a.length);
console.log(\\"all.js(3:0)\\", a.map());

function add() {
  console.log(\\"all.js(6:2)\\", \\"add:\\", \\"1\\");
}

var subtract = function subtract() {
  console.log(\\"all.js(10:2)\\", \\"subtract:\\", \\"2\\");
};

var multiply = void 0;
multiple = function multiple() {
  console.log(\\"all.js(15:2)\\", \\"multiple:\\", \\"3\\");
};

var divide = void 0;
divide = function divide() {
  console.log(\\"all.js(20:2)\\", \\"divide:\\", \\"4\\");
};

var power = function pow() {
  console.log(\\"all.js(24:2)\\", \\"power:\\", 5);
};

var obj = {
  method1: function method1() {
    console.log(\\"all.js(29:4)\\", \\"method1:\\", 6);
  },
  method2: function method2() {
    console.log(\\"all.js(32:4)\\", \\"method2:\\", 7);
  },
  method3: function method3() {
    console.log(\\"all.js(35:4)\\", \\"method3:\\", 8);
  }
};

var Component = function Component() {
  var privateMethod = function privateMethod() {
    console.log(\\"all.js(41:4)\\", \\"Component.privateMethod:\\", 1);
  };
  return function () {
    function _class2() {
      (0, _classCallCheck3.default)(this, _class2);

      this.showList = function () {
        arr.map(function (i) {
          console.log(\\"all.js(46:8)\\", \\"i\\", i);
        });
      };
    }

    (0, _createClass3.default)(_class2, [{
      key: \\"render\\",
      value: function render() {
        console.log(\\"all.js(50:6)\\", \\"value:\\", 2);
      }
    }]);
    return _class2;
  }();
};"
`;
